### PR TITLE
Update bdk-swift local build script

### PR DESCRIPTION
**Pull request by notmandatory**
### Description

While @reez was testing this script he found some errors. This should fix them.

#### All Submissions:

* [x] I've signed all my commits
* [x] I followed the [contribution guidelines](https://github.com/bitcoindevkit/bdk/blob/master/CONTRIBUTING.md)
* [x] I ran `cargo fmt` and `cargo clippy` before committing

### CODE REVIEW

**Comment by notmandatory** [comment]
Hey @ytemiloluwa you will need these fixes also to test out the build script on your machine.

**Comment by Ytemiloluwa** [comment]
> Hey @ytemiloluwa you will need these fixes also to test out the build script on your machine.

Okay Got it @notmandatory

File: bdk-swift/build-local-swift.sh [hunk 1]
```diff
@@ -5,25 +5,24 @@
 #
 # Run the script from the repo root directory, ie: ./bdk-swift/build-local-swift.sh
 
-rustup install nightly-x86_64-apple-darwin
-rustup component add rust-src --toolchain nightly-x86_64-apple-darwin
+rustup install nightly-2023-04-10
+rustup component add rust-src --toolchain nightly-2023-04-10
 rustup target add aarch64-apple-ios x86_64-apple-ios
-rustup target add aarch64-apple-ios-sim --toolchain nightly
+rustup target add aarch64-apple-ios-sim --toolchain nightly-2023-04-10
```
Comment (reez) [inline bdk-swift/build-local-swift.sh:11]: removing `x86_64-apple-darwin` helps resolve the error/warnings I was getting:

```
error: DEPRECATED: future versions of rustup will require --force-non-host to install a non-host toolchain as the default.

warning: toolchain 'nightly-2023-04-10-x86_64-apple-darwin' may not be able to run on this system.

warning: If you meant to build software to target that platform, perhaps try `rustup target add x86_64-apple-darwin` instead?
```

File: bdk-swift/build-local-swift.sh [hunk 2]
```diff
@@ -5,25 +5,24 @@
...
 rustup target add aarch64-apple-darwin x86_64-apple-darwin
 
 pushd bdk-ffi
-mkdir -p Sources/BitcoinDevKit
```
Comment (reez) [inline bdk-swift/build-local-swift.sh:15]: confirmed that removing this line did not affect anything with getting the successful build, and could be removed.

**Review by reez — APPROVED** [review]
Awesome! Thanks for making these changes, they look great, and run successfully for me.

**Comment by thunderbiscuit** [comment]
Should any of this make its way into bdk-swift? Or are those fixes only for the local builds and don't affect the CI build?

**Comment by notmandatory** [comment]
> Should any of this make its way into bdk-swift? Or are those fixes only for the local builds and don't affect the CI build?

These changes came from the `bdk-swift` :-) .

**Review by thunderbiscuit — APPROVED** [review]
ACK 38a40f54a169ac58c0b132ab65798ccc585b2582.

**Comment by thunderbiscuit** [comment]
Mind rebasing this @notmandatory?

**Comment by thunderbiscuit** [comment]
I tried rebasing this but there are merge conflicts and I didn't want to mess anything up... I think it's ready to go though.

**Pull request merged**